### ISSUE TITLE
Compiling with nw-gyp?

### ISSUE THREAD

**Issue by reggi** [TooTallNate/NodObjC#52]
Hey Nate!

I have no experience with using `node-gyp` or building a node module. I'm confused as to if there is a source that isn't in the repo. 

I attempted to rebuild `NodObjC` but I kept running into errors (no `binding.gyp` file).

Anyway! In order to get a module [osx-background](https://github.com/hughsk/osx-background) working with [node-webkit](https://github.com/rogerwang/node-webkit) I need a version of NodObjC that is compiled with [https://github.com/rogerwang/nw-gyp](nw-gyp).

This should fix the issue I'm having https://github.com/rogerwang/node-webkit/issues/1277#issuecomment-61324540.

Rodger the maintainer of both `nw-gyp` and `node-webkit` [states that](https://github.com/rogerwang/node-webkit/issues/520#issuecomment-14598247) NodObjC needs to be built for node-webkit natively.

https://github.com/rogerwang/node-webkit/wiki/Build-native-modules-with-nw-gyp

But this is still an icky issue because then I'll have to create a new versions of both modules `webkit-NodObjC` and `webkit-osx-background` or is their some way during `npm install` that I can specify which to use `node-gyp` or `nw-gyp`?

Sorry to bug you with this, I saw somewhere that you stated this should be fixed on the `node-webkit` end of things?

**Comment by TooTallNate** [comment]
@reggi Sorry for late response. Did you ever get this figured out?

**Comment by MarcFowler** [comment]
No, I never had any luck with the (current) version of node-webkit when I tried recently. There has been some movement over in the other thread though: https://github.com/nwjs/nw.js/issues/1811#issuecomment-68829397 I don't know if this will help though, I'll check as soon as I can get to it. Looks like a fork of node-ffi to help?

**Comment by reggi** [comment]
@TooTallNate @MarcFowler I never figured out a solution for using the module I needed I just made another script (a workaround) for setting the background in osx using `node-osascript`. Plus I moved to `atom-shell`, which had the issue as well.

```
var osascript = require('node-osascript');

module.exports = osxBackgroundGet;

function osxBackgroundGet(callback) {
  osascript.execute([
    'tell application ``System Events"',
    'tell current desktop',
    'get picture',
    'end tell',
    'end tell'
  ].join("\n"), callback);
}
```

But I could still see this being an issue with some modules & node-webkit / atom-shell.

**Comment by hlian1** [comment]
Just felt like I need to say that NodObjC builds fine with nw-gyp using nwjs 0.12.2. Only ref and ffi need to be rebuilt. The example "Hello World" application then works. Seems like  `$.NSApplication('sharedApplication')` returns a new application; I was not able to get nwjs's own `NSApp`.
